### PR TITLE
Added code for Round Robin Scheduling

### LINKED ISSUE #1873: Add code for Round Robin CPU Scheduling
I would like to add the implementation of this in cpp. Please assign this.
@raghav-dalmia  @MastersAbh  @asha15  @singh-shreya6  @SakshiShreya @somya-kapoor

### PR DESCRIPTION
**Pull request by Xxoliaxx**
Fixes #1873

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request: 
- Added the implementation of Round Robin scheduling in c++

#### Languages Used:
- C++

#### Files added:
- Round_Robin_Scheduling.cpp

### CODE REVIEW

**Comment by SakshiShreya** [comment]
Please mention the issue number in the first line of the PR template after #. And complete the template properly. A lot of things are missing.

**Comment by Xxoliaxx** [comment]
I have made the necessary changes. Do check and review. @SakshiShreya

**Comment by raghav-dalmia** [comment]
your pr show 2 file change, remove the irrelevant ones. Also, squash all the commits. Rest seems good to me.

**Comment by Xxoliaxx** [comment]
@raghav-dalmia I removed the irrelevant file but I am having trouble squashing the commits since I am not well versed in it. Do help.

File: Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#include<iostream>
+using namespace std;
+
+int main()
+{        
+	int i, j, no_of_process, current_time, remaining_time, f=0, time_slice, wait_time=0, turn_time=0;
+    cout<<"Enter number of processes: ";
```
Comment (singh-shreya6) [inline Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp]: Please fix the indentation.

File: Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    cin>>no_of_process;
+
+    int a[no_of_process];	//Arrival Time
+	int b[no_of_process];	//Burst Time
```
Comment (singh-shreya6) [inline Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp]: indentation fix

File: Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
+#include<iostream>
+using namespace std;
+
+int main()
+{        
+	int i, j, no_of_process, current_time, remaining_time, f=0, time_slice, wait_time=0, turn_time=0;
+    cout<<"Enter number of processes: ";
+    cin>>no_of_process;
```
Comment (singh-shreya6) [inline Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp]: space around >>

File: Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    int a[no_of_process];	//Arrival Time
+	int b[no_of_process];	//Burst Time
+	int r[no_of_process];
+    remaining_time = no_of_process;
+
+    for(i=0;i<no_of_process;i++)
```
Comment (singh-shreya6) [inline Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp]: Please give spaces around all operators

File: Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+             if(r[i] == 0 && f == 1)
+             {        remaining_time--;
+                      cout<<"\nProcess "<<i<<"\t"<<current_time-a[i]<<"\t\t"<<current_time-a[i]-b[i];
+                      wait_time + = current_time-a[i]-b[i];
+                      turn_time + = current_time-a[i];
+                      f = 0;
+             }
+
+             if(i > = no_of_process-1)
```
Comment (singh-shreya6) [inline Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp]: no space between >= as it is a single operator

**Comment by jainaman224** [comment]
@Xxoliaxx Are you working on it?

**Comment by Xxoliaxx** [comment]
@jainaman224 Yes I am working on it.

**Comment by Xxoliaxx** [comment]
@singh-shreya6 I have made the necessary changes

File: Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+            i++;
+        else
+            i = 0;
+     }
+     cout << "\nAverage Waiting Time: " << (wait_time*1.0) / no_of_process;
+     cout << "\nAverage Turnaround Time: " << (turn_time*1.0) / no_of_process;
+     return 0;
+}
+
+Sample Input/Output:
```
Comment (singh-shreya6) [inline Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp]: Please add these inside multiline comments.

File: Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp [hunk 7]
```diff
@@ -0,0 +1,88 @@
+#include<iostream>
```
Comment (singh-shreya6) [inline Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp]: Add brief problem description on the top as comment

**Comment by Xxoliaxx** [comment]
@singh-shreya6 I have made the requested changes.

File: Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp [hunk 8]
```diff
@@ -0,0 +1,101 @@
+//Round Robin Scheduling is a CPU scheduling algorithm where each process is assigned a fixed time slot in a cyclic way.
```
Comment (singh-shreya6) [inline Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp]: This single line comment style does not look neat. Please add the entire thing inside a multi line comment.

File: Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+  Burst Time: Time required by a process for CPU execution.
+  Turn Around Time: Time Difference between completion time and arrival time.          
+  Waiting Time: Time Difference between turn around time and burst time. */
+
+#include<iostream>
+using namespace std;
+
+int main()
+{        
+    int i, j, no_of_process, current_time, remaining_time, f = 0, time_slice, wait_time = 0, turn_time = 0;
```
Comment (singh-shreya6) [inline Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp]: split this across two lines

File: Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    cout << "Enter Time slice: ";
+    cin >> time_slice;
+
+    cout << "\nProcess\tTurnaround time\t Waiting Time\n";
+    cout << "-------------------------------------------";
+    for(current_time = 0, i = 0; remaining_time != 0;)
+    {
+        if(r[i] <= time_slice && r[i] > 0)	//If burst time is less than time slice
```
Comment (singh-shreya6) [inline Scheduling_Algorithm/C++/Preemptive/Round_Robin_Scheduling.cpp]: Please add the comments before the lines rather than on the sides.

**Comment by Xxoliaxx** [comment]
@singh-shreya6 do check the changes if they're right

**Pull request merged**